source/common/http/conn_pool_base.h

```diff
@@ -78,8 +83,13 @@ class ActiveClient : public Envoy::ConnectionPool::ActiveClient {
                uint64_t concurrent_stream_limit)
       : Envoy::ConnectionPool::ActiveClient(parent, lifetime_stream_limit,
                                             concurrent_stream_limit) {
-    Upstream::Host::CreateConnectionData data = parent_.host()->createConnection(
-        parent_.dispatcher(), parent_.socketOptions(), parent_.transportSocketOptions());
+    Upstream::Host::CreateConnectionData data =
+        static_cast<Envoy::ConnectionPool::ConnPoolImplBase*>(&parent)->host()->createConnection(
```

Comment:
What's going on with this cast? Isn't this an upcast, and thus would happen implicitly?

Reply:
Annoyingly, 
HttpConnPoolImplBase has
Upstream::HostDescriptionConstSharedPtr host() const override { return host_; }

and the base class has
const Upstream::HostConstSharedPtr& host() const { return host_; }

so without the cast it snags the wrong type and complains there's no createConnection function.

Reply:
Yuck. Can you add a comment to that effect?

Reply:
yeah - will add it to my clean up list too - may be able to do away with it with one of the later refactors!